### PR TITLE
Use Names::getValidNameGivenExisting in binary reading

### PR DESCRIPTION
**Pull request by kripken**
We had a TODO to use it once Names was optimized, which it has been.

The Names version is also far faster. When building
https://github.com/JetBrains/kotlinconf-app it saves 70 seconds(!).

### CODE REVIEW

**Comment by kripken** [comment]
This isn't NFC as there are minor observable differences in the names compared to before, but the name names should be just as good.

Added a test for collisions with the new naming scheme.

File: test/duplicated_names_collision_underscore.wasm
Comment (tlively) [inline test/duplicated_names_collision_underscore.wasm:1]: Can you add this as a lit test under lit/binary instead? Although it looks a like there is already an updated lit test, so maybe it isn't necessary to add a new test at all?

**Review by tlively — APPROVED** [review]
Wow, that's a shocking improvement.

File: test/duplicated_names_collision_underscore.wasm
Comment (kripken) [reply test/duplicated_names_collision_underscore.wasm:1]: The tests check slightly different things, different possible name overlaps, with '.', '_', '$', so I think we need all three. I converted the non-lit ones to lit.

**Pull request merged**